mobile/lib/modules/home/services/asset_cache.service.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:http/http.dart';
+import 'package:openapi/api.dart';
+import 'package:path_provider/path_provider.dart';
+
+abstract class JsonCache<T> {
```

Comment:
Should we put this abstract class to a separate file in the shared module? Since this class is used in both AlbumCacheService and AssetCacheService?

Reply:
Oh yeah, I planned to refactor this but forgot about it :D

Reply:
Done.